packages/dd-trace/src/span_processor.js

```diff
@@ -0,0 +1,37 @@
+const log = require('./log')
+
+class SpanProcessor {
+  constructor (exporter, prioritySampler) {
+    this._exporter = exporter
+    this._prioritySampler = prioritySampler
+  }
+
+  process (span) {
+    const spanContext = span.context()
+    const trace = spanContext._trace
+
+    if (trace.started.length === trace.finished.length) {
+      this._prioritySampler.sample(spanContext)
+
+      if (spanContext._traceFlags.sampled === false) {
+        log.debug(() => `Dropping trace due to user configured filtering: ${trace}`)
+        this._erase(trace)
+        return
+      }
+      this._exporter.export(span)
```

Comment:
I think this way plays nicely if we want to give the span processor multiple exporters to submit to. That way, the exporters don't clobber each other by trying to both modify the spans, it's left up to the processor to do it.

Reply:
What I meant is that `format()` is called in the exporter instead of in the processor. I wonder if that should be done in the processor instead. This will be important when we implement user-defined hooks into the processor since at that point they should have a formatted span to work with before it reaches the exporters.

Reply:
Okay, that makes sense